internal/sourcemap/body_caching_test.go

```diff
@@ -0,0 +1,167 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sourcemap
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/go-sourcemap/sourcemap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-server/internal/elasticsearch"
+)
+
+var unsupportedVersionSourcemap = `{
+  "version": 1,
+  "sources": ["webpack:///bundle.js"],
+  "names": [],
+  "mappings": "CAAS",
+  "file": "bundle.js",
+  "sourcesContent": [],
+  "sourceRoot": ""
+}`
+
+func Test_NewCachingFetcher(t *testing.T) {
+	_, err := NewBodyCachingFetcher(nil, -1, nil)
+	require.Error(t, err)
+
+	f, err := NewBodyCachingFetcher(nil, 100, nil)
+	require.NoError(t, err)
+	assert.NotNil(t, f.cache)
+}
+
+func TestStore_Fetch(t *testing.T) {
```

Comment:
I'm struggling to follow this test. Why does it test anything to do with Elasticsearch? AFAICS the BodyCachingFetcher is generic, so could we use a mock backend Fetcher which exercises the caching logic? Rather than updating the cache directly, make a call that would be uncached, have it hit the backend (mock) fetcher, and return a specified nil/non-nil consumer. Then you can check that the caching fetcher does the right thing.

EDIT: I see this is pre-existing code that's been moved around :)
I suppose it's fine for now, but something we should come back and improve.